### PR TITLE
feat: add arabic layout

### PR DESCRIPTION
**Pull request by Oderjunkie**
this definitely isn't _especially_ polished, can't really lie about that, but i thought i'd share it. the bigrams are probably wack though

### CODE REVIEW

**Comment by FoxMoss** [comment]
lgtm

**Comment by flide** [comment]
#344

File: 8vim/src/main/res/raw/ar.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            top:
+              - lower_case: "َ"
+              - lower_case: "ً"
+              - lower_case: "آ"
+              - lower_case: "إ"
+            bottom:
+              - lower_case: "ُ"
+              - lower_case: "ٌ"
+              - lower_case: "’"
+              - lower_case: null
```
Comment (MaethorNaur) [inline 8vim/src/main/res/raw/ar.yaml]:
```suggestion
```

File: 8vim/src/main/res/raw/ar.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        bottom:
+          parts:
+            right:
+              - lower_case: "ى"
+              - lower_case: null
+              - lower_case: null
+              - lower_case: null
```
Comment (MaethorNaur) [inline 8vim/src/main/res/raw/ar.yaml]:
```suggestion
```

File: 8vim/src/main/res/raw/ar.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            left:
+              - lower_case: "ـ"
+              - lower_case: null
+              - lower_case: null
+              - lower_case: null
```
Comment (MaethorNaur) [inline 8vim/src/main/res/raw/ar.yaml]:
```suggestion
```

File: 8vim/src/main/res/raw/ar.yaml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        left:
+          parts:
+            bottom:
+              - lower_case: "ِ"
+              - lower_case: "ٍ"
+              - lower_case: null
+              - lower_case: null
```
Comment (MaethorNaur) [inline 8vim/src/main/res/raw/ar.yaml]:
```suggestion
```

File: 8vim/src/main/res/raw/ar.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            top:
+              - lower_case: "ّ"
+              - lower_case: "٠"
+              - lower_case: "١"
+              - lower_case: null
```
Comment (MaethorNaur) [inline 8vim/src/main/res/raw/ar.yaml]:
```suggestion
```

**Review by MaethorNaur — CHANGES_REQUESTED** [review]
The layout is invalid: https://github.com/8VIM/8VIM/actions/runs/5639605374/job/15275073453?pr=339

**Comment by MaethorNaur** [comment]
It shouldn't be `- lower_case: null` but just `- null`.
Also, those lines are unnecessary as they are at the end of the quadrant and won't be rendered by default.

Setting `- null` can be helpful if you want to skip a position, like:

```yaml
left:
  - null
  - lower_case: a
  - lower_case: b
 ```

We skip the first position to render `a` and `b` at the 2nd and 3rd positions

**Review by Oderjunkie — COMMENTED** [review]
ah. i was in a hurry and Forgot to delete those oops

**Review by MaethorNaur — APPROVED** [review]

**Pull request merged**